examples/bc5cdr.py

```diff
@@ -183,45 +227,99 @@ def _split_generators(self, dl_manager):
+        db_id = span.infons[db_id_key] if db_id_key else -1
+
         return {
+            "id": span.id,
             "offsets": offsets,
-            "text": span.text,
-            "type": span.infons['type'],
-            "entity_id": kb_id,
+            "text": [span.text],
+            "type": span.infons["type"],
+            "normalized": [{"db_name": db_id_key, "db_id": db_id}],
         }
 
     def _generate_examples(
         self,
         filepath,
         split,  # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
     ):
-        """Yields examples as (key, example) tuples."""
-        if self.config.name == "biocreative":
+        """ Yields examples as (key, example) tuples. """
+        if self.config.name == "source":
             reader = bioc.BioCXMLDocumentReader(str(filepath))
-            for id_, xdoc in enumerate(reader):
-                yield id_, {
+            for uid, xdoc in enumerate(reader):
+                yield uid, {
                     "passages": [
                         {
-                            "pmid": xdoc.id,
-                            "type": passage.infons['type'],
+                            "document_id": xdoc.id,
+                            "type": passage.infons["type"],
                             "text": passage.text,
                             "entities": [
                                 self._get_bioc_entity(span)
                                 for span in passage.annotations
                             ],
                             "relations": [
                                 {
-                                    key.lower(): value
-                                    for key, value in rel.infons.items()
+                                    "id": rel.id,
+                                    "type": rel.infons["relation"],
+                                    "arg1_id": rel.infons["Chemical"],
+                                    "arg2_id": rel.infons["Disease"],
                                 }
                                 for rel in xdoc.relations
                             ],
                         }
                         for passage in xdoc.passages
                     ]
                 }
+
+        elif self.config.name == "bigbio":
+            reader = bioc.BioCXMLDocumentReader(str(filepath))
+            uid = 0  # global unique id
+
+            for i, xdoc in enumerate(reader):
+                data = {
+                    "id": uid,
+                    "document_id": xdoc.id,
+                    "passages": [],
+                    "entities": [],
+                    "relations": [],
+                }
+                uid += 1
+
+                char_start = 0
+                # passages must not overlap and spans must cover the entire document
+                for passage in xdoc.passages:
+                    offsets = [[char_start, char_start + len(passage.text)]]
```

Comment:
don't need nested iterables for passage offsets and its incompatible with the single string field for `text`

Reply:
Again. matching for consistency per Slack